xc/device_common.ml

```diff
@@ -334,15 +334,20 @@ let is_upstream_qemu domid =
 		with_xs (fun xs -> xs.Xs.read (Printf.sprintf "/libxl/%d/dm-version" domid)) = "qemu_xen"
 	with _ -> false
 
-let qmp_write domid cmd =
+let qmp_write domid ?(read_result=false) cmd  =
+	let open Qmp in
 	let open Qmp_protocol in
 	let c = connect (Printf.sprintf "/var/run/xen/qmp-libxl-%d" domid) in
+	let result = ref (Success (None, Unit)) in
 	finally
 		(fun() ->
 			try
 				negotiate c;
-				write c cmd
+				write c cmd;
+				if read_result then
+					result := Qmp_protocol.read c;
+				!result
```

Comment:
Also, it could even be better to make the response `optional` or even `result`, to allow you to know if some error happened.

```ocaml
(* or qmp_write_with_response or something on those lines *)
let qmp_write_and_read domid cmd  =
  let open Qmp in
  let open Qmp_protocol in
  let c = connect (Printf.sprintf "/var/run/xen/qmp-libxl-%d" domid) in
  finally
    (fun () ->
      try
        negotiate c;
        write c cmd;
        if read_result then 
          Some (Qmp_protocol.read c)
        else None
      with e ->
        error "Caught exception attempting to write qmp message: %s" (Printexc.to_string e);
        None)
    (fun () -> close c)

let qmp_write domid cmd  = qmp_write_and_read domid cmd |> ignore
```

This way, if you get an exception you know.